lib/compress.js

```diff
@@ -2130,9 +2130,6 @@ merge(Compressor.prototype, {
                     extract_declarations_from_unreachable_code(compressor, self.body, a);
                     return make_node(AST_BlockStatement, self, { body: a });
                 }
-            } else {
-                // self instanceof AST_Do
-                return self.body;
```

Comment:
But removing those lines would be a clean revert. That `return make_node(AST_For, self, self).transform(compressor);` part from down below has always been there, just tucked away in `OPT(AST_While)` instead of here instead. So adding `return self;` would actually alter the flow.

See also https://github.com/mishoo/UglifyJS2/pull/1534#issuecomment-283702822